.github/workflows/build-sample-apps.yml

```diff
@@ -0,0 +1,126 @@
+name: Build sample apps 
+
+on: 
+  pull_request: # build sample apps for every commit pushed to an open pull request (including drafts)
+  push: 
+    branches: [main] 
+  release: # build sample apps for every git tag created. These are known as "stable" builds that are suitable for people outside the mobile team. 
+    types: [published]
+
+concurrency: # cancel previous workflow run if one exists. 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  update-pr-comment:
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write # to be able to comment on PR
+    outputs:
+      comment-id: ${{ steps.create-comment.outputs.comment-id }}
+    steps:
+    - name: Find Comment
+      uses: peter-evans/find-comment@v2
+      id: existing-comment
+      with:
+        issue-number: ${{ github.event.pull_request.number }}
+        comment-author: 'github-actions[bot]'
+        body-includes: <!-- sample app builds --> 
+    
+    - name: Create or update comment
+      uses: peter-evans/create-or-update-comment@v3
+      id: create-comment
+      with:
+        comment-id: ${{ steps.existing-comment.outputs.comment-id }}
+        issue-number: ${{ github.event.pull_request.number }}
+        body: |
+          <!-- sample app builds --> 
+          # Sample app builds 📱
+          
+          Below you will find the list of the latest versions of the sample apps. It's recommended to always download the latest builds of the sample apps to accurately test the pull request. 
+
+          ---
+          ${{ steps.build.outputs.build-log }}
+        edit-mode: replace # replace the existing comment with new content since we are creating new builds 
```

Comment:
Not sure what is the best way, but I feel like updating comments seems too big of a hassle considering we might be adding more information to it?.

What do you think about consolidating the build information into a single comment? This will require adding a step to create an artifact that stores the build information and then reading from that artifact to generate the PR comment at the end. 

so instead of incremental changes, we just do one comment at the end? 

In future, if we have to add something, we don't go about changing the comment again, but rather add something to that artifact. 

something like, 
```
const fs = require('fs');
          const buildInfos = fs.readdirSync('.').filter(name => name.startsWith('build-info'));
```

Reply:
Sorry, I don't understand your question or concern? 

The way this workflow is setup today is: 
* When a new commit is pushed to a PR, delete the old PR comment and replace with a new blank comment saying that builds are starting. 
* After each build is created, they separately update the PR comment with a success or failed message for that 1 build. 
* This PR comment is only meant to be for sample apps compiling. If we want to automate more then building sample apps in the future, we can create separate PR comments for that purpose.

Reply:
It's not a blocker, just observation. My concern was about the part 

> After each build is created, they separately update the PR comment with a success or failed message for that 1 build.

I believe we can just add a comment once we have a success or failure, or we can keep storing its state in some artifacts and once both apps processing is done then make a comment. 

> we can create separate PR comments for that purpose.

I think if we keep on adding/updating comments like these, they might lose value. So I prefer fewer automated comments on PR. So we know what information each comment is going to have. I think that way it would be easier to maintain. 

Like I said earlier, these are just my thought no actions step need to come out of this, if more folks on the team doesnt feel the same.